client.py

```diff
@@ -249,30 +253,86 @@ def format_day_counts(self, bucket: list):
     def format_counts(self, bucket: list):
         return {item["key"]: item["doc_count"] for item in bucket}
 
-    def search_overview(self, collection: str, q: str):
+    def aggregator_query(
+        self, collection: str, q: str, *aggs: QueryBuilder.Aggregators, **options
+    ):
         """
-        Get overview statistics for a query
+        Abstraction to DRY out permutations of the 'overview' query getting broken out into their own calls
         """
-        res = self.ES.search(index=collection, body=QueryBuilder(q).overview_query())  # type: ignore [call-arg]
+        query_body = QueryBuilder(q).aggregator_query(*aggs)
+
+        res = self.ES.search(index=collection, body=query_body)  # type: ignore [call-arg]
         if not res["hits"]["hits"]:
             raise HTTPException(status_code=404, detail="No results found!")
 
         total = res["hits"]["total"]["value"]
-        tldsum = sum(
-            item["doc_count"] for item in res["aggregations"]["tld"]["buckets"]
-        )
-        return {
+        return_dict = {
             "query": q,
-            "total": max(total, tldsum),
-            "topdomains": self.format_counts(res["aggregations"]["domain"]["buckets"]),
-            "toptlds": self.format_counts(res["aggregations"]["tld"]["buckets"]),
-            "toplangs": self.format_counts(res["aggregations"]["lang"]["buckets"]),
-            "dailycounts": self.format_day_counts(
-                res["aggregations"]["daily"]["buckets"]
-            ),
-            "matches": [self.format_match(h, collection) for h in res["hits"]["hits"]],
         }
 
+        # Add the results of each aggregator to the return value
+        for agg in aggs:
+            agg_name = next(iter(agg.value.keys()))
```

Comment:
A bit hard to grok ... are we interested in anything more than the first item in the list returned by `keys()`?

Reply:
I feel like I come upon this specific syntactic wrinkle recurrently - you're right, we only want the first item returned by keys- so I guess `list(agg.value.keys())[0]` would also work and be easier to grok

Reply:
fixed this